src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -183,14 +209,25 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
         }
       }
 
-      if (!Strings.isNullOrEmpty(kmsKeyName)) {
-        loadConfig.setDestinationEncryptionConfiguration(new EncryptionConfiguration().setKmsKeyName(kmsKeyName));
-      }
+      temporaryTableReference = null;
+      if (tableExists && !isTableEmpty(tableRef) && !Operation.INSERT.equals(operation)) {
+        String temporaryTableName = UUID.randomUUID().toString().replaceAll("-", "_");
+        temporaryTableReference = new TableReference()
+          .setDatasetId(tableRef.getDatasetId())
+          .setProjectId(tableRef.getProjectId())
+          .setTableId(temporaryTableName);
+        loadConfig.setDestinationTable(temporaryTableReference);
+      } else {
+        loadConfig.setDestinationTable(tableRef);
+        if (!Strings.isNullOrEmpty(kmsKeyName)) {
```

Comment:
this should be done regardless of the operation type. If they want the final data to be encrypted, we should also encrypt the intermediate data.

Reply:
Done.